libfreerdp/core/proxy.c

```diff
@@ -253,3 +257,136 @@ BOOL http_proxy_connect(BIO* bufferedBio, const char* hostname, UINT16 port)
 	return TRUE;
 }
 
+/* SOCKS Proxy auth methods by rfc1928 */
+#define AUTH_M_NO_AUTH   0
+#define AUTH_M_GSSAPI    1
+#define AUTH_M_USR_PASS  2
+
+static BOOL socks_proxy_connect(BIO* bufferedBio, const char* hostname, UINT16 port)
+{
+	int status;
+	BYTE buf[280], hostnlen = 0xff & strlen(hostname);
+	/* CONN REQ replies in enum. order */
```

Comment:
usually we would have written `strlen(hostname) & 0xff`

Reply:
Why not.